File: Middle-logic/Models/Session.js
export class Session {
    constructor(Session_ID,meditator, startDateTime, endDateTime, practicedStages = [],  ahaMoments = [], steps = [],newlyMasteredStages = []) {
        this.Session_ID = Session_ID;
        this.Meditator = meditator;
        this.Start_Date_Time = startDateTime;
        this.End_Date_Time = endDateTime;
        this.Practiced_Stages = practicedStages;
        this.AhaMoments = ahaMoments;
        this.Steps = steps;
        this.Newly_Mastered_Stages = newlyMasteredStages;
    }
    static getSessionFromObject(obj){
        return new Session(
            obj.Session_ID,
            obj.Meditator,
            obj.Start_Date_Time,
            obj.End_Date_Time,
            obj.Practiced_Stages,
            obj.AhaMoments,
            obj.Steps,
            [],
        );
    }
    static convertArrayToSessionObjects(arr){
        var SessionObjects= arr.map(session=>
            new Session(session.Session_ID,session.Meditator,session.Start_Date_Time,session.End_Date_Time,session.Practiced_Stages,session.AhaMoments,session.Steps)
        )
        return SessionObjects;
    }

}

File: backend/Models/Session.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public class Session
    {
        public int Id { get; set; }
        public DateTime StartDateTime { get; set; }
        public DateTime EndDateTime { get; set; }

        [Column("MeditatorID")]
        public Meditator Meditator { get; set; }
        public List<AhaMoment> AhaMoments { get; set; }
        public List<PracticedStage>? PracticedStages { get; set; }
        public List<NewlyMasteredStage>? NewlyMasterdStages { get; set; }
        public List<ObservableObject> ObservableObjects { get; set; }
        public PreparationPhase PreparationPhase { get; set; }
        public int PreparationPhaseId { get; set; } // Foreign key property
        public bool IsDeleted { get; set; } = false;
        public Session()
        {

        }
        public Session(DateTime startDateTime, DateTime endDateTime, Meditator meditator, List<AhaMoment> ahaMoments,
            List<PracticedStage> practicedStages, List<NewlyMasteredStage> newlyMasteredStages, List<ObservableObject> observableObjects)
        {
            StartDateTime = startDateTime;
            EndDateTime = endDateTime;
            Meditator = meditator;
            AhaMoments = ahaMoments;
            PracticedStages = practicedStages;
            ObservableObjects = observableObjects;
            NewlyMasterdStages = newlyMasteredStages;
        }

    }
}


